helper-cli/src/main/kotlin/commands/ConvertOrtFileCommand.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2022 EPAM Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.helper.commands
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.convert
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.file
+import org.ossreviewtoolkit.helper.common.readOrtResult
+import org.ossreviewtoolkit.helper.common.writeOrtResult
+import org.ossreviewtoolkit.utils.common.expandTilde
+
+class ConvertOrtFileCommand: CliktCommand(
+    help = "Converts the given ORT file to a different format, e.g. '.json' to '.yml'."
+) {
+    private val inputOrtFile by option(
+        "--input-ort-file", "-i",
+        help = "The ORT file to convert."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val outputOrtFile by option(
+        "--output-ort-file", "-o",
+        help = "The ORT file to write to."
```

Comment:
It probably makes sense to state here explicitly that the file extension used for the output file determines the output format.

I assume the use case to have untypical file extensions is not worth to be supported, right? This would be possible with a `--target-format` option that takes precedence over the file extension.

Reply:
This reminds me that we already have a generic transform helper command, or? Would it make sense to instead extend that command to use an identity transformation by default, and to determine the target format via the file extension?

Reply:
This didn't quite work out. My use case was converting 'ort/cli/src/funTest/assets/semver4j-analyzer-result.yml' to 'json' and then reading it via `org.ossreviewtoolkit.helper.common.UtilsKt.readOrtResult()`. It runs into the following issue, while using the mapper for (de-)serializing it works.

``` 
Exception in thread "main" com.fasterxml.jackson.databind.exc.InvalidTypeIdException: Could not resolve subtype of [simple type, class org.ossreviewtoolkit.model.config.ScanStorageConfiguration]: missing type id property '@c' (for POJO property 'storages')
 at [Source: (File); line: 234, column: 9]
```

Reply:
I ended up reverting back to the previous approach with a separate command as the transform command seems not to utilize ORT's custom serialization code which does not fit the needs.

Reply:
ISTR that the transform command had some limitations anyway regarding the support of enhanced YAML features, but don't know any more for sure. So I am fine with reverting.